FEXCore/Source/Utils/Allocator/HostAllocator.h

```diff
@@ -46,5 +50,5 @@ namespace Alloc {
 namespace Alloc::OSAllocator {
 void RegisterTLSData(FEXCore::Core::InternalThreadState *Thread);
 void UninstallTLSData(FEXCore::Core::InternalThreadState *Thread);
-fextl::unique_ptr<Alloc::HostAllocator> Create64BitAllocator();
+fextl::unique_ptr<Alloc::HostAllocator> Create64BitAllocator(fextl::vector<FEXCore::Allocator::MemoryRegion> *MemoryRegion = nullptr);
```

Comment:
This looks really poorly designed. Previously this function had reasonably obvious behavior, but with this change you really have open up the implementation to see what's going on. Can you come up with a more predictable interface that doesn't use a heavy `std::vector` in-out parameter?